src/firefly/js/visualize/saga/CoverageWatcher.js

```diff
@@ -0,0 +1,427 @@
+        }
+
+
+        switch (action.type) {
+
+            case TABLE_NEW:
+                if (!getTableInGroup(tbl_id)) continue;
+                decimatedTables[tbl_id]= null;
+                displayedTableId = updateCoverage(tbl_id, viewerId, decimatedTables, options);
+                break;
+
+            case TBL_RESULTS_ACTIVE:
+                if (!getTableInGroup(tbl_id)) continue;
+                displayedTableId = updateCoverage(tbl_id, viewerId, decimatedTables, options);
+                break;
+
+            case TABLE_REMOVE:
+                if (!getTableInGroup(payload.tbl_id)) continue;
+                removeCoverage(payload.tbl_id, decimatedTables);
+                displayedTableId = null;
+                previousDisplayedTableId = null;
+                tbl_id = getActiveTableId();
+                if (!isEmpty(decimatedTables)) {
+                    if (!tbl_id) tbl_id= Object.keys(decimatedTables)[0]; // todo - remove this line when getActiveTable() gets fixed
+                    displayedTableId = updateCoverage(tbl_id, viewerId, decimatedTables, options);
+                }
+                break;
+
+            case MultiViewCntlr.ADD_VIEWER:
+            case MultiViewCntlr.VIEWER_MOUNTED:
+                if (action.payload.viewerId === viewerId) {
+                    paused = false;
+                    tbl_id = getActiveTableId();
+                    displayedTableId = updateCoverage(tbl_id, viewerId, decimatedTables, options);
+                }
+                break;
+
+            case MultiViewCntlr.VIEWER_UNMOUNTED:
+                if (action.payload.viewerId === viewerId) paused = true;
+                break;
+            case ImagePlotCntlr.PLOT_IMAGE:
+                if (action.payload.plotId===PLOT_ID) overlayCoverageDrawing(decimatedTables,options);
+                break;
+        }
+        if (!displayedTableId) displayedTableId= previousDisplayedTableId;
+    }
+}
+
+
+
+function removeCoverage(tbl_id, decimatedTables) {
+    Reflect.deleteProperty(decimatedTables, tbl_id);
+    if (!Object.keys(decimatedTables)) {
+        dispatchDeletePlotView(PLOT_ID);
+    }
+}
+
+
+/**
+ * 
+ * @param tbl_id
+ * @param viewerId
+ * @param decimatedTables
+ * @param options
+ * @return {Array}
+ */
+function updateCoverage(tbl_id, viewerId, decimatedTables, options) {
+
+    if (!tbl_id) return null;
+    const table= getTblById(tbl_id);
+    if (!table) return null;
+    if (!options.hasCoverageData(options, table)) return null;
+    if (decimatedTables[tbl_id]==='WORKING') return tbl_id;
+
+
+    const params= {
+        startIdx : 0,
+        pageSize : 1000000,
+        inclCols : getCovColumnsForQuery(options, table)
+    };
+
+    const req = Object.assign({}, table.request, params);
+    req.tbl_id = tbl_id;
+
+    if (decimatedTables[tbl_id] /*&& decimatedTables[tbl_id].tableMeta.source===table.tableMeta.source*/) { //todo support decimated data
```

Comment:
tableMeta.source is also fixed, but it may not be what you want.  A sort will produce a new source.